scripts/setup/generate_secrets.py

```diff
@@ -180,6 +184,20 @@ def add_secret(name: str, value: str) -> None:
     if need_secret("zulip_org_id"):
         add_secret("zulip_org_id", str(uuid.uuid4()))
 
+    if need_secret("push_registration_encryption_keys"):
```

Comment:
Will this result generating these on any Zulip server, or just on the bouncer server? I think `need_secret` would generate this on every server, but I wasn't sure if that was intended.

Reply:
OK, the thing I was missing is that we may need these for a development environment. We're never going to "generate secrets during installation" for Zulip Cloud, which is the production deployment of this system.

I think you can have this code check the `development` and only generate this secret there and that'll resolve this thread.

Reply:
Made the changes:
```py
    if development and need_secret("push_registration_encryption_keys"):
        private_key = PrivateKey.generate()
        private_key_str = Base64Encoder.encode(bytes(private_key)).decode("utf-8")
        public_key_str = Base64Encoder.encode(bytes(private_key.public_key)).decode("utf-8")
        add_secret(
            "push_registration_encryption_keys", json.dumps({public_key_str: private_key_str})
        )
```

Reply:
OK. Can you also add a comment:

> settings.ZILENCER_ENABLED would be a better check than development for whether we need push bouncer secrets, but we're trying to avoid importing settings.

Reply:
Comment added.